src/core/misc.c

```diff
@@ -207,13 +207,15 @@ void gslist_free_full (GSList *list, GDestroyNotify free_func)
 	g_slist_free(list);
 }
 
-GSList *gslist_remove_string (GSList *list, const char *str)
+GSList *gslist_delete_string (GSList *list, const char *str, GDestroyNotify free_func)
```

Comment:
`gslist_remove_string` was API and is still in misc.h

Reply:
Indeed, but the old one was wrong and potentially leaky.
We could re-add it in a correct form using `g_slist_delete_link` instead or re-write it in terms of `gslist_delete_string` (with a `NULL` `free_func`, with minor modifications to `gslist_delete_string`).

Reply:
I see, but only LemonBoy was using this api